pkg/pb/overlay.proto

```diff
@@ -85,6 +86,16 @@ message QueryResponse {
 message PingRequest {};
 message PingResponse {};
 
+// TODO: add fields that validate who is requesting the info
+message InfoRequest {}
+
+message InfoResponse {
+    bytes id = 1 [(gogoproto.customtype) = "NodeID", (gogoproto.nullable) = false];
```

Comment:
This shouldn't be here. We cannot trust the other side giving valid response. Use `PeerIdentityFromPeer` to get the identity.

Reply:
I don't understand. The ID is just one of the "info" properties returned by the node. So if the sender has just the address of the target node, it will receive the target's ID, type, metadata and restrictions. I don't understand how this relates with the peer identity.

Reply:
Because the responder can write anything it wants to that field.

Reply:
The only thing we can trust is the NodeID and certificate that it's using for communicating with us. Because, we can verify that it actually has the corresponding private key and certificate.

Reply:
OK. I see the issue. I will remove the ID from the response.

Reply:
Done.